docs/src/pages/playground.tsx

```diff
@@ -0,0 +1,212 @@
+	const [embedLoading, setEmbedLoading] = useState(true)
+	const { colorMode } = useColorMode()
+	const [fetchError, setFetchError] = useState<null | string>(null)
+	const [templates, setTemplates] = useState<Project[] | null>(null)
+	const [selectedTemplateIndex, setSelectedTemplateIndex] = useState(0)
+
+	const embedDiv = useRef<HTMLIFrameElement | null>(null)
+	const editor = useRef(null)
+
+	useEffect(() => {
+		const fetchTemplates = async () => {
+			const res = await fetch('http://localhost:3000/api/templates')
+
+			if (res.status !== 20) {
+				setFetchError(res.statusText)
+			}
+
+			if (res.ok && res.status === 200) {
+				const json = await res.json()
+				setTemplates(json)
+			}
+		}
+		fetchTemplates()
+	}, [])
+
+	useEffect(() => {
+		if (editor && editor.current) {
+			editor.current.setTheme(colorMode)
+		}
+	}, [colorMode])
+
+	const clickHandler = () => {
+		if (dropdown) {
+			setDropdown(false)
+		}
+	}
+	useEffect(() => {
+		document.addEventListener('click', clickHandler)
+
+		return () => {
+			document.removeEventListener('click', clickHandler)
+		}
+	}, [])
+
+	useEffect(() => {
+		if (embedDiv && embedDiv.current) {
+			sdk
+				.embedGithubProject('embed', projectLink, {
+					devToolsHeight: 40,
+					showSidebar: true,
+					terminalHeight: 40,
+					theme: colorMode,
+					view: 'default'
+				})
+				.then((v) => {
+					setEmbedLoading(false)
+					editor.current = v.editor
+				})
+		}
+	}, [projectLink])
+
+	function renderSearchBar() {
+		if (!embedLoading) {
+			if (dropdown) {
+				return (
+					<SearchbarFocused
+						data={templates}
+						setProjectLink={setProjectLink}
+						setDropdown={setDropdown}
+						setSelectedTemplateIndex={setSelectedTemplateIndex}
+						selectedTemplateIndex={selectedTemplateIndex}
+					/>
+				)
+			} else {
+				return (
+					<input
+						className={styles.searchBar}
+						onFocus={() => {
+							if (fetchError !== null) {
+								setDropdown(true)
+							}
+						}}
+						placeholder={fetchError !== null ? 'Search for a Robo template' : fetchError}
+					></input>
+				)
+			}
+		}
+	}
+
+	return (
+		<>
+			<div className={styles.searchBarContainer} style={{ height: dropdown ? '27px' : 'unset' }}>
+				{renderSearchBar()}
+			</div>
+			<div ref={embedDiv} id="embed" onClick={clickHandler} />
+		</>
+	)
+}
+
+interface SearchBarProps {
+	data: Project[] | null
+	setProjectLink: React.Dispatch<React.SetStateAction<string>>
+	setDropdown: React.Dispatch<React.SetStateAction<boolean>>
+	selectedTemplateIndex: number
+	setSelectedTemplateIndex: React.Dispatch<React.SetStateAction<number>>
+}
+
+function SearchbarFocused(props: SearchBarProps) {
+	const { data, setProjectLink, setDropdown, setSelectedTemplateIndex, selectedTemplateIndex } = props
+	const { colorMode } = useColorMode()
+	const [searchTemplate, setSearchTemplate] = useState('')
+
+	const ResultsRender = () => {
+		if (data !== null) {
+			return data.map((project: Project, _: number) => {
```

Comment:
Holy nested Batman! Is there a way you could flatten this structure a bit?

Reply:
I can try to, but it includes extra data just in case we want to separate things even more, I ll try to make it cleaner but keep the data intact, unless you think we wont ever need any of Categories or sub categories